.github/actions/setup-postgres/action.yml

```diff
@@ -0,0 +1,62 @@
+name: Setup PostgreSQL
+description: |
+  Setup a PostgreSQL for Linux, macOS and Windows machines. Runs natively, no
+  Docker containers are involved.
+inputs:
+  username:
+    description: A username of the user to setup.
+    default: postgres
+  password:
+    description: A password of the user to setup.
+    default: postgres
+  database:
+    description: A database name to setup and grant permissions to created user.
+    default: postgres
+runs:
+  using: composite
+  steps:
+    - name: Prerequisites
+      run: |
+        if [ "$RUNNER_OS" == "Linux" ]; then
+          echo "$(pg_config --bindir)" >> $GITHUB_PATH
+        elif [ "$RUNNER_OS" == "Windows" ]; then
+          echo "$PGBIN" >> $GITHUB_PATH
+          echo "PQ_LIB_DIR=$PGROOT\lib" >> $GITHUB_ENV
+        fi
+      shell: bash
+
+
+    - name: Setup and start PostgreSQL
+      run: |
+        export PGDATA="$RUNNER_TEMP/pgdata"
+        pg_ctl init
```

Comment:
Do you know if there is an easy way for us to check the contents of `$PGDATA/postgresql.conf` at this point? (maybe just `cat` it to stdout?). I wonder what `listen_addresses` is set to (https://www.postgresql.org/docs/13/runtime-config-connection.html#GUC-LISTEN-ADDRESSES). Now that we've disabled unix sockets, it would be nice to double check that we do not allow remote connections and only listen on 127.0.0.1 and/or ::1. Or am I being paranoid and Github Actions already enforce strict network isolation?

Reply:
I'm not sure what default is, it could be 0.0.0.0. You should be able to see the listening address in logs since Postgres prints them on start.

I could explicitly set it to 127.0.0.1 though.

Reply:
From logs:

```
2021-01-16 22:58:27.988 UTC [3484] LOG:  listening on IPv6 address "::1", port 5432
2021-01-16 22:58:27.989 UTC [3484] LOG:  listening on IPv4 address "127.0.0.1", port 5432
```

BTW, even if unix sockets are enabled, tcp/ip sockets are there anyway. My understanding is that tooling such as `createuser` and `psql` were using unix sockets by default, and the path to these sockets has been set compile time (unless overriden via `PGHOST` env var).

Reply:
Thank you for confirming! :+1: 

> BTW, even if unix sockets are enabled, tcp/ip sockets are there anyway. My understanding is that tooling such as createuser and psql were using unix sockets by default, and the path to these sockets has been set compile time (unless overriden via PGHOST env var).

Right. It can get confusing at times, as they will also use unix sockets when you use `localhost` in the DB connection string.